### PR TITLE
Add C# example for the AudioStreamGenerator code snippet

### PR DESCRIPTION
**Pull request by NiftyHat**
Added a C# example for the AudioStreamGenerator code snippet.

### CODE REVIEW

File: doc/classes/AudioStreamGenerator.xml [hunk 1]
```diff
@@ -24,6 +25,39 @@
 		    for i in range(frames_available):
 		        playback.push_frame(Vector2.ONE * sin(phase * TAU))
 		        phase = fmod(phase + increment, 1.0)
+		[/gdscript]
+		[csharp]
+                [Export] public AudioStreamPlayer Player { get; set; }
```
Comment (akien-mga) [inline doc/classes/AudioStreamGenerator.xml:30]: The codeblock should be indented with tabs like above (and then indentation of the code itself within the codeblock is made with 4 spaces).

File: doc/classes/AudioStreamGenerator.xml [hunk 2]
```diff
@@ -24,6 +25,39 @@
...
+
+                private float _pulseHz = 440.0f; // The frequency of the sound wave.
+                private float _sampleHz;
+                private AudioStreamGeneratorPlayback _playback; //Will hold the AudioStreamGeneratorPlayback.
```
Comment (akien-mga) [inline doc/classes/AudioStreamGenerator.xml:34]:
```suggestion
                private AudioStreamGeneratorPlayback _playback; // Will hold the AudioStreamGeneratorPlayback.
```

Comment (akien-mga) [inline doc/classes/AudioStreamGenerator.xml:34]: I would keep the order consistent with GDScript, this should be a 1:1 translation if possible.
If the order used here seems better, the GDScript example can be changed too.

File: doc/classes/AudioStreamGenerator.xml [hunk 3]
```diff
@@ -24,6 +25,39 @@
...
+
+                public override void _Ready()
+                {
+                    base._Ready();
```
Comment (akien-mga) [inline doc/classes/AudioStreamGenerator.xml:38]: In GDScript it's not calling `super()`, so this seems unnecessary.

File: doc/classes/AudioStreamGenerator.xml [hunk 4]
```diff
@@ -24,6 +25,39 @@
...
+                    if (Player.Stream is AudioStreamGenerator generator) //need to type as a generator so you can access MixRate
```
Comment (akien-mga) [inline doc/classes/AudioStreamGenerator.xml:39]:
```suggestion
                    if (Player.Stream is AudioStreamGenerator generator) // Type as a generator to access MixRate.
```

**Comment by NiftyHat** [comment]
I've made the changes but I think something went wrong the the amendment and now there are two commits instead of one. I did try to squash them but git wouldn't let me due to the commits having multiple parents :(

**Comment by akien-mga** [comment]
I squashed the commits for you, it wasn't trivial indeed after the latest changes, not sure how the Git history ended up exactly.

I also fixed a few formatting issues (trailing whitespace, `[codeblocks]` should be plural when using the language-specific subblocks, and `<` needs to be escaped as `&lt;` in XML).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**